app/components/github_integration/webhooks/issues.py

```diff
@@ -0,0 +1,169 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Literal, Protocol, cast
+
+from app.components.github_integration.webhooks.core import (
+    EmbedContent,
+    Footer,
+    SubhookStore,
+    client,
+    make_subhook_registrar,
+    send_embed,
+)
+
+if TYPE_CHECKING:
+    from githubkit.typing import Missing
+    from githubkit.versions.v2022_11_28.models import (
+        WebhookIssueCommentCreated,
+        WebhookIssuesClosed,
+        WebhookIssuesLocked,
+        WebhookIssuesOpened,
+        WebhookIssuesPinned,
+        WebhookIssuesReopened,
+        WebhookIssuesUnlocked,
+        WebhookIssuesUnpinned,
+    )
+    from monalisten.types import IssueCommentEvent, IssuesEvent
+
+    from app.components.github_integration.emoji import EmojiName
+
+EXPECTED_STATE_REASONS = frozenset({"completed", "not_planned", "duplicate"})
+
+issue_subhooks: SubhookStore[IssuesEvent] = {}
+
+register_issue_subhook = make_subhook_registrar(issue_subhooks)
+
+
+class IssueLike(Protocol):
+    state: Missing[Literal["open", "closed"]]
+    state_reason: Missing[str | None]
+
+
+def get_issue_emoji(issue: IssueLike) -> EmojiName:
+    if issue.state == "open":
+        return "issue_open"
+    suffix = "completed" if issue.state_reason == "completed" else "unplanned"
+    return "issue_closed_" + suffix
+
+
+@client.on("issues")
+async def handle_issue_event(event: IssuesEvent) -> None:
+    if subhook := issue_subhooks.get(event.action):
+        await subhook(event)
+
+
+@register_issue_subhook("opened")
+async def handle_opened_issue(event: WebhookIssuesOpened) -> None:
+    issue, number = event.issue, event.issue.number
+    await send_embed(
+        event.sender,
+        EmbedContent(f"opened Issue #{number}", issue.html_url, issue.body),
+        Footer("issue_open", f"Issue #{number}: {issue.title}"),
+        color="green",
+    )
+
+
+@register_issue_subhook("closed")
+async def handle_closed_issue(event: WebhookIssuesClosed) -> None:
+    issue, number = event.issue, event.issue.number
+    if issue.state_reason not in EXPECTED_STATE_REASONS:
+        print("Unexpected state reason in handle_closed_issue:", issue.state_reason)
+        return
+
+    if issue.state_reason == "completed":
+        color, emoji_kind = "purple", "completed"
+    else:
+        color, emoji_kind = "gray", "unplanned"
```

Comment:
In my opinion `EXPECTED_STATE_REASONS` should be inlined since one is already special-cased here and there are only two others.

```suggestion
    if issue.state_reason == "completed":
        color, emoji_kind = "purple", "completed"
    elif issue.state_reason in ("not_planned", "duplicate"):
        color, emoji_kind = "gray", "unplanned"
    else:
        print("Unexpected state reason in handle_closed_issue:", issue.state_reason)
        return
```

(Or maybe a match-case if you consider that to be more readable.)